src/helm/proxy/clients/together_client.py

```
@@ -72,28 +87,84 @@ def __init__(self, cache_config: CacheConfig, api_key: Optional[str] = None):
-            except Exception as e:
-                raise TogetherClientError(
-                    f"Together request failed with {response.status_code}: {response.text}"
-                ) from e
-            result = response.json()
-            return result["output"]
-
-        def fail():
-            raise RuntimeError(f"The result has not been uploaded to the cache for the following request: {cache_key}")
-
-        response, cached = self.cache.get(cache_key, wrap_request_time(do_it if self.api_key else fail))
+        if not self.api_key:
+            raise TogetherClientError("togetherApiKey not set in credentials.conf")
+        headers: Dict[str, str] = {"Authorization": f"Bearer {self.api_key}"}
+
+        if request.model_engine in _ASYNC_MODELS:
+
+            def submit_job() -> str:
+                submit_request = raw_request.copy()
+                submit_request["async"] = True
+                submit_response = requests.post(TogetherClient.INFERENCE_ENDPOINT, headers=headers, json=submit_request)
+                try:
+                    submit_response.raise_for_status()
+                except Exception as e:
+                    raise TogetherClientError(
+                        f"Together job submission request failed with {submit_response.status_code}: "
+                        f"{submit_response.text}"
+                    ) from e
+                submit_response_json = submit_response.json()
+                job_id = submit_response_json.get("id")
+                if not job_id:
+                    raise TogetherClientError(
+                        f"Could not get job_id from job submission response {submit_response_json}"
+                    )
+                return job_id
+
+            # Retry with a 1 second delay that doubles every retry until a maximum of delay of 15 seconds.
+            # Stop retrying after 1 minute.
+            @retry(wait_exponential_multiplier=1000, wait_exponential_max=15000, stop_max_delay=60000)
```

Does the retry respond to any Exception?  Could we be more selective about this to avoid swallowing up exceptions unintentionally?

---

Made it only retry `JobNotFinishedError`.